fs/any_version_test.go

```diff
@@ -0,0 +1,82 @@
+package fs
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/hc-install/product"
+)
+
+func TestAnyVersionValidate(t *testing.T) {
+	t.Parallel()
+
+	productWithInvalidBinaryName := product.Terraform
+	productWithInvalidBinaryName.BinaryName = func() string { return "invalid!" }
+
+	testCases := map[string]struct {
+		av          AnyVersion
+		expectedErr error
+	}{
+		"no-fields-set": {
+			av:          AnyVersion{},
+			expectedErr: fmt.Errorf("must use either ExactBinPath or Product + ExtraPaths"),
+		},
+		"ExactBinPath-and-Product": {
+			av: AnyVersion{
+				ExactBinPath: "/test",
+				Product:      &product.Terraform,
+			},
+			expectedErr: fmt.Errorf("use either ExactBinPath or Product + ExtraPaths, not both"),
+		},
+		"ExactBinPath-empty": {
+			av: AnyVersion{
+				ExactBinPath: "",
+			},
+			expectedErr: fmt.Errorf("must use either ExactBinPath or Product + ExtraPaths"),
+		},
+		"ExactBinPath-relative": {
+			av: AnyVersion{
+				ExactBinPath: "test",
+			},
+			expectedErr: fmt.Errorf("expected ExactBinPath (\"test\") to be an absolute path"),
+		},
+		"ExactBinPath-absolute": {
+			av: AnyVersion{
+				ExactBinPath: "/test",
```

Comment:
```suggestion
				ExactBinPath: filepath.Separator + "test",
```

This should fix the failing Windows test, hopefully.

Reply:
❤️  Thank you! Saved me from looking that one up.

Reply:
Or you could probably have a separate `_windows_test.go` which has the path beginning with `C:\` to be more realistic, but I'm not sure if that's necessary.

Reply:
Looks like this will need to be a more "real" absolute Windows path 😑

Reply:
Maybe a little heavy, but going to try `filepath.Join(t.TempDir(), "test")`